subsys/net/ip/tcp.c

```diff
@@ -870,9 +870,16 @@ int net_tcp_send_pkt(struct net_pkt *pkt)
 	net_pkt_cursor_init(pkt);
 	net_pkt_set_overwrite(pkt, true);
 
-	if (net_pkt_skip(pkt, net_pkt_ip_hdr_len(pkt) +
-			 net_pkt_ipv6_ext_len(pkt))) {
-		return -EMSGSIZE;
+	if (net_pkt_family(pkt) == AF_INET) {
+		if (net_pkt_skip(pkt, net_pkt_ip_hdr_len(pkt) +
+				 net_pkt_ipv4_opts_len(pkt))) {
+			return -EMSGSIZE;
+		}
+	} else if (net_pkt_family(pkt) == AF_INET6) {
+		if (net_pkt_skip(pkt, net_pkt_ip_hdr_len(pkt) +
+				 net_pkt_ipv6_ext_len(pkt))) {
+			return -EMSGSIZE;
+		}
```

Comment:
Here it would avoid the 2 if on AF_INET/AF_INET6.  you would just run:

if (net_pkt_skip(pkt, net_pkt_ip_hdr_len(pkt) + net_pkt_ip_opts_len(pkt))) { ... }

less code, simple, easier to read etc...